web/src/stream_popover.js

```diff
@@ -509,19 +477,19 @@ export function register_stream_handlers() {
     });
 
     // Choose a different color.
-    $("body").on("click", ".choose_stream_color", (e) => {
-        update_spectrum($(e.target).closest(".streams_popover"), ($colorpicker) => {
-            $(".colorpicker-container").show();
-            $colorpicker.spectrum("destroy");
-            $colorpicker.spectrum(stream_color.sidebar_popover_colorpicker_options_full);
-            // In theory this should clean up the old color picker,
-            // but this seems a bit flaky -- the new colorpicker
-            // doesn't fire until you click a button, but the buttons
-            // have been hidden.  We work around this by just manually
-            // fixing it up here.
-            $colorpicker.parent().find(".sp-container").removeClass("sp-buttons-disabled");
-            $(e.target).hide();
-        });
+    $("body").one("click", ".choose_stream_color", (e) => {
+        const $popover = $(e.target).closest(".streams_popover");
+        const $colorpicker = $popover.find(".colorpicker-container").find(".colorpicker");
+        $(".colorpicker-container").show();
+        $colorpicker.spectrum("destroy");
+        $colorpicker.spectrum(stream_color.sidebar_popover_colorpicker_options_full);
+        // In theory this should clean up the old color picker,
+        // but this seems a bit flaky -- the new colorpicker
+        // doesn't fire until you click a button, but the buttons
+        // have been hidden.  We work around this by just manually
+        // fixing it up here.
+        $colorpicker.parent().find(".sp-container").removeClass("sp-buttons-disabled");
+        $(e.target).hide();
```

Comment:
The color picker CSS seems broken in the night theme:

![image](https://github.com/zulip/zulip/assets/2746074/c3e92150-57b6-4ab9-b729-838e83f40256)

Reply:
This might not be the best fix, but it works:

``` diff
diff --git a/web/styles/popovers.css b/web/styles/popovers.css
index 5b94ea1efb..b3a8df4e3f 100644
--- a/web/styles/popovers.css
+++ b/web/styles/popovers.css
@@ -4,6 +4,11 @@
   popovers. (Tippy tooltips are defined in tooltips.css).
 */
 .tippy-box[data-theme="light-border"] {
+    .sp-input {
+        /* Override incorrect color for text in dark theme. */
+        color: var(--color-text-default) !important;
+    }
+
     .tippy-content {
         font-size: 14px;
     }
```